framework/source/class/qx/ui/window/Window.js

```diff
@@ -632,17 +679,34 @@ qx.Class.define("qx.ui.window.Window",
         this.hide();
         this.fireEvent("close");
       }
+
+      // If automatically destroying the window upon close was requested, do
+      // so now. (Note that we explicitly re-obtain the autoDestroy property
+      // value, allowing the user's close handler to enable/disable it before
+      // here.)
+      if (this.getAutoDestroy()) {
+        this.removeAll();                                    // remove contents
+        qx.event.Registration.removeAllListeners(this);      // remove listeners
+        qx.core.Init.getApplication().getRoot().remove(this);// remove self
+        this.dispose();
+      }
     },
```

Comment:
For some reason we did not do it this way and used the *onDisappear* event to handle that. I don't recall the reason, because it's so much time has passed now. Will check when I'm back @ work...